spec/java/util/List.main.lsl

```diff
@@ -0,0 +1,542 @@
+import java/util/stream/_interfaces;
+
+
+// local semantic types
+
+@implements("java.util.List")
+@public @interface type CustomList
+    is java.util.List
+    for List
+{
+    //@private @static val serialVersionUID: long = 1; // #problem: should be 3905348978240129619
+}
+
+
+// automata
+
+automaton CustomListAutomaton
+(
+)
+: CustomList
+{
+    // states and shifts
+
+    initstate Initialized;
+
+    shift Initialized -> self by [
+        // static operations
+        copyOf,
+        of (),
+        of (Object),
+        of (Object, Object),
+        of (Object, Object, Object),
+        of (Object, Object, Object, Object),
+        of (Object, Object, Object, Object, Object),
+        of (Object, Object, Object, Object, Object, Object),
+        of (Object, Object, Object, Object, Object, Object, Object),
+        of (Object, Object, Object, Object, Object, Object, Object, Object),
+        of (Object, Object, Object, Object, Object, Object, Object, Object, Object),
+        of (Object, Object, Object, Object, Object, Object, Object, Object, Object, Object),
+        of (array<Object>),
+    ];
+
+    shift Initialized -> self by [
+        // instance methods
+        add (CustomList, Object),
+        add (CustomList, int, Object),
+        addAll (CustomList, Collection),
+        addAll (CustomList, int, Collection),
+        clear,
+        contains,
+        containsAll,
+        equals,
+        forEach,
+        get,
+        hashCode,
+        indexOf,
+        isEmpty,
+        iterator,
+        lastIndexOf,
+        listIterator (CustomList),
+        listIterator (CustomList, int),
+        parallelStream,
+        remove (CustomList, Object),
+        remove (CustomList, int),
+        removeAll,
+        removeIf,
+        replaceAll,
+        retainAll,
+        set,
+        size,
+        sort,
+        spliterator,
+        stream,
+        subList,
+        toArray (CustomList),
+        toArray (CustomList, IntFunction),
+        toArray (CustomList, array<Object>),
+    ];
+
+    // internal variables
+
+    // utilities
+
+    // constructors
+
+    // static methods
+
+    @static fun *.copyOf (coll: Collection): List
+    {
+        val data: list<Object> = action LIST_NEW();
+
+        var size: int = 0;
+        val iter: Iterator = action CALL_METHOD(coll, "iterator", []);
+        action LOOP_WHILE(
+            action CALL_METHOD(iter, "hasNext", []),
+            copyOf_loop(iter, data, size)
+        );
+
+        result = new ArrayListAutomaton(state = Initialized,
+            storage = data,
+            length = size,
+        );
+    }
+
+    @Phantom proc copyOf_loop (iter: Iterator, data: list<Object>, size: int): void
+    {
+        val item: Object = action CALL_METHOD(iter, "next", []);
+        action LIST_INSERT_AT(data, size, item);
+        size += 1;
+    }
+
+
+    @static fun *.of (): List
+    {
+        result = new ArrayListAutomaton(state = Initialized,
```

Comment:
Why it returns original ArrayListAutomaton ? If we will try to add any element in this List - all will be ok (as I understand); But this is ImmutableList;

Reply:
[UtBot](https://github.com/UnitTestBot/UTBotJava/blob/main/utbot-framework/src/main/java/org/utbot/engine/overrides/collections/List.java#L10) sources were used as primary examples.